lib/opte/src/engine/dhcpv6/protocol.rs

```diff
@@ -321,9 +323,8 @@ fn generate_reply_options<'a>(
 
     // If requested, provide the list of DNS servers.
     if msg.has_option_request_with(OptionCode::DnsServers) {
-        let opt = Dhcpv6Option::DnsServers(IpList::from(
-            action.dns_servers.as_slice(),
-        ));
+        let ip_list = IpList(Cow::Owned(action.dhcp_cfg.dns6_servers.clone()));
```

Comment:
Nit: can probably just use `Cow::Borrowed` here since the `dhcp_cfg` isn't behind a RwLock anymore.